src/service/resources-impl.js

```diff
@@ -49,6 +50,16 @@ const MUTATE_DEFER_DELAY_ = 500;
 const FOCUS_HISTORY_TIMEOUT_ = 1000 * 60;  // 1min
 const FOUR_FRAME_DELAY_ = 70;
 
+/**
+ * The internal state of the visibility state machine.
+ * @typedef {{
+ *   viewer: boolean,
```

Comment:
This definition is no longer correct, right? Also, do we need a struct now at all?